tests/test_run_subproc.py

```diff
@@ -0,0 +1,47 @@
+import os
+import sys
+import builtins
+
+import pytest
+
+from xonsh.platform import ON_WINDOWS
+from xonsh.built_ins import run_subproc
+
+from tools import skip_if_on_windows
+
+
+@pytest.yield_fixture(autouse=True)
+def chdir_to_test_dir(xonsh_builtins):
```

Comment:
Well, the purpose of this is to get the xonsh/tests/bin dir on the relative path, not to create or modify files. The redirect one below does do some of that, but I think that this is fine for now.  A refactor of that would be nice.